### PR TITLE
Use `true` for RACE_ENABLED

### PR DESCRIPTION
**Pull request by 6543**
as title

### CODE REVIEW

**Comment by zeripath** [comment]
I may have gotten the wrong end of the stick here but I think holding the mutex should be being locked and unlocked before reading from the archiveInProgress list. In fact the archiver code states that any access to this should be locked.

If so the correct fix is not to hold the mutex throughout that function but to release and then relock in the for loop. e.g.:

```go
func releaseOneEntry(t *testing.T, inFlight []*ArchiveRequest) {
	var nowQueued, numQueued int

	archiveMutex.Lock()
	numQueued = len(archiveInProgress)
	archiveMutex.Unlock()

	// Release one, then wait up to 10 seconds for it to complete.
	queueMutex.Lock()
	archiveQueueReleaseCond.Signal()
	queueMutex.Unlock()

	timeout := time.Now().Add(10 * time.Second)
	for {
		archiveMutex.Lock()
		nowQueued = len(archiveInProgress)
		archiveMutex.Unlock()
		if nowQueued != numQueued || time.Now().After(timeout) {
			break
		}
		runtime.Gosched()
		<-time.After(50 * time.Millisecond)
	}

	// Make sure we didn't just timeout.
	assert.NotEqual(t, numQueued, nowQueued)

	// Also make sure that we released only one.
	assert.Equal(t, numQueued-1, nowQueued)
}
```

Please also note the <-time.After and the gosched call as the current loop keeps polling that array!

**Comment by 6543** [comment]
```
--- FAIL: TestArchive_Basic (3.05s)
    archiver_test.go:179: 
                Error Trace:    archiver_test.go:179
                Error:          Should be true
                Test:           TestArchive_Basic
FAIL
FAIL    code.gitea.io/gitea/services/archiver   3.400s
FAIL
```
:point_up: only thing that has to be fixed and it's no more wip

**Comment by zeripath** [comment]
ok I think you might be now overlocking.

If there a value never changes - then there doesn't need to be a mutex to read it.

For example, `archivePath` is set only once at `ArchiveRequest` construction in `DeriveRequestFrom`. Therefore there cannot be a race from reading it.

**Comment by zeripath** [comment]
Similary `ArchiveRequest.uri`, `ArchiveRequest.repo`,  `ArchiveRequest.refname`,   `ArchiveRequest.ext`,   `ArchiveRequest.archivePath`,  `ArchiveRequest.archiveType`.

However, `ArchiveRequest.complete` does get changed in multiple places so - this is one which potentially does need locking - and you can see on line 279 there is a read from `request.archiveComplete` that is outside of a lock - so depending on how `getArchiveRequest` works and if the `ArchiveRequest` it returns is not a copy made under lock then that read does need a surrounding lock.

**Comment by 6543** [comment]
since #14723 will do a way better job, lets just take the refactor of Makefile and merge it in ...

cc @lunny 

PS: I'll ajust #1441 after it got merged ...

File: Makefile [hunk 1]
```diff
@@ -64,8 +64,9 @@ EXTRA_GOFLAGS ?=
 MAKE_VERSION := $(shell $(MAKE) -v | head -n 1)
 MAKE_EVIDENCE_DIR := .make_evidence
 
-ifneq ($(RACE_ENABLED),)
-	GOTESTFLAGS ?= -race
+ifneq ($(RACE_ENABLED),"true")
```
Comment (6543) [inline Makefile:67]:
```suggestion
ifeq ($(RACE_ENABLED),"true")
```

**Review by lunny — APPROVED** [review]

**Review by KN4CK3R — APPROVED** [review]

**Pull request merged**